doc/userguide/devguide/internals/engines/stream/applayertriggerrawreassembly.rst

```diff
@@ -0,0 +1,37 @@
+AppLayerParserTriggerRawStreamReassembly
+----------------------------------------
+
+This is a call for immediate reassembly of raw stream data by the Application Layer in case
+there is a requirement e.g. upon completion of an HTTP request.
+
+The stream engine provides the following parameters for the stream to be tracked by the different
+subsystems of Suricata. These parameters are used to track the progress of the stream in
+the respective subsystems. This serves as a communication between the different subsystems
+about how far they have already looked into the stream.
+They are all relative to the base offset of the streaming buffer region.
+
+* app layer progress
+* raw reassembly progress
+* streaming logger progress
+
+For a given streaming buffer as follows and a raw stream progress as seen by the detection
+engine
+
+.. image:: stream-engine/trigger_reassembly_before.png
+
+Note that there could be any amount of data in the stream but we're only concerned about the
+last ACK'd data.
```

Comment:
Suggestion to move this to after the explanation, so the thought flow is not cut in the middle.